### ISSUE TITLE
`cmd+enter` is not working on Mac

### ISSUE THREAD

**Issue by MrJGao** [Ikuyadeu/vscode-R#482]
**Describe the bug**
The `cmd+enter` to run the seletion/line is not working on Mac, even if removing the conflict keybindings.

**To Reproduce**
1. Remove the "insert new line" keybinding.
2. Restart vscode.
3. Now, the `cmd+enter` should work, but if you delete the current terminal and create a new one, the `cmd+enter` shortcut won't work again.

I got 2 MacBook pros having this issue. Thanks.

**Comment by andycraig** [comment]
Hi @MrJGao, thank you for reporting this. 

> Now, the cmd+enter should work, but if you delete the current terminal and create a new one, the cmd+enter shortcut won't work again.

When you try this, does <kbd>Cmd-Enter</kbd> do anything? Is there an error? Or does nothing happen at all?

Can you copy-paste the contents of your `settings.json` file and `keybindings.json` file here?

Thank you.

**Comment by MrJGao** [comment]
Thanks. when I do `cmd+enter`, it moves the cursor to the next line. 

here's my `settings.json`:

```
{
    // "r.rterm.windows": "C:\\Program Files\\R\\R-3.6.2\\bin\\R.exe",
    "r.rterm.windows": "C:\\Users\\xgao26\\AppData\\Local\\Continuum\\anaconda3\\Scripts\\radian.exe",
    "r.rterm.mac": "/Users/xgao26/Library/Python/3.7/bin/radian",
    "window.zoomLevel": 0,
    "python.jediEnabled": false,
    "workbench.sideBar.location": "left",
    "editor.renderControlCharacters": true,
    "editor.renderWhitespace": "boundary",
    "workbench.panel.defaultLocation": "right",
    "files.associations": {
        "*.rmd": "r",
        "*.Rmd": "rmd"
    },
    "[r]": {
        "editor.defaultFormatter": "REditorSupport.r-lsp"
        
    },
    "editor.formatOnPaste": true,
    "[rmd]": {
        "editor.rulers": [85]
    },
    "sync.gist": "3aaaef13503bc8777c0e2adefbd8b7a2",
    "r.editor.tabSize": 4,
    "code-runner.saveFileBeforeRun": true,
    "r.linting.doubleQuotes": false,
    "git.autofetch": true,
    "r.linting.camelCase": true,
    "terminal.integrated.shell.windows": "C:\\WINDOWS\\System32\\cmd.exe",
    "editor.minimap.enabled": false,
    "git.enableSmartCommit": true,
    "breadcrumbs.enabled": true,
    "git.ignoreLegacyWarning": true,
    "liveServer.settings.donotShowInfoMsg": true,
    "liveServer.settings.donotVerifyTags": true,
    "zenMode.centerLayout": false,
    "zenMode.hideLineNumbers": false,
    "zenMode.hideTabs": false,
    "files.autoSave": "off",
    "workbench.iconTheme": "vscode-icons",
    "markdown.extension.preview.autoShowPreviewToSide": true,
    "bracket-pair-colorizer-2.colors": [
        "White",
        "Gold",
        "Orchid",
        "LightSkyBlue"
    ],
    "bracket-pair-colorizer-2.colorMode": "Independent",
    "better-comments.tags": [
        {
            "tag": "!",
            "color": "#FF2D00",
            "strikethrough": false,
            "backgroundColor": "transparent"
        },
        {
            "tag": "?",
            "color": "#3498DB",
            "strikethrough": false,
            "backgroundColor": "transparent"
        },
        {
            "tag": "//",
            "color": "#474747",
            "strikethrough": true,
            "backgroundColor": "transparent"
        },
        {
            "tag": "todo",
            "color": "#FF8C00",
            "strikethrough": false,
            "backgroundColor": "transparent"
        },
        {
            "tag": "*",
            "color": "#82E0AA",
            "strikethrough": false,
            "backgroundColor": "transparent"
        },
        {
            "tag": "~",
            "color": "#82E0AA",
            "strikethrough": false,
            "backgroundColor": "transparent"
        },
        {
            "tag": "===",
            "color": "#d62eb8",
            "strikethrough": false,
            "backgroundColor": "transparent"
        },
        {
            "tag": "region",
            "color": "white",
            "strikethrough": false,
            "backgroundColor": "#08504a"
        },
        {
            "tag": "endregion",
            "color": "white",
            "strikethrough": false,
            "backgroundColor": "#383737"
        },
        {
            "tag": "fig:",
            "color": "white",
            "backgroundColor": "purple"
        }
    ],
    "sync.autoUpload": true,
    "sync.autoDownload": true,
    "sync.quietSync": true,
    "todo-tree.tree.showScanModeButton": true,
    "todo-tree.general.tags": [
        "BUG",
        "HACK",
        "FIXME",
        "TODO",
        "XXX",
        "NOTE",
        "fig:",
        "out:"
    ],
    "beautify.config": "",
    "r.bracketedPaste": true,
    // when need to debug on HPC, use these
    // "r.bracketedPaste": false,
    "todo-tree.highlights.customHighlight": {
        "NOTE" : {
            "foreground": "black",
            "iconColour": "#D4AC0D",
            "background": "#D4AC0D"
        },
        "out:": {
            "foreground": "black",
            "iconColour": "#65e098",
            "background": "#65e098"
        }
    },
    "python.languageServer": "Microsoft",
    "python.dataScience.sendSelectionToInteractiveWindow": true,
    "python.defaultInterpreterPath": "/usr/bin/python3",
    "r.lsp.diagnostics": false,
    "hediet.vscode-drawio.local-storage": "eyIuZHJhd2lvLWNvbmZpZyI6IntcImxhbmd1YWdlXCI6XCJcIixcImN1c3RvbUZvbnRzXCI6W10sXCJsaWJyYXJpZXNcIjpcImdlbmVyYWw7YXJyb3dzMlwiLFwiY3VzdG9tTGlicmFyaWVzXCI6W1wiTC5zY3JhdGNocGFkXCJdLFwicGx1Z2luc1wiOltdLFwicmVjZW50Q29sb3JzXCI6W1wiNkU2RTZFXCJdLFwiZm9ybWF0V2lkdGhcIjoyNDAsXCJjcmVhdGVUYXJnZXRcIjpmYWxzZSxcInBhZ2VGb3JtYXRcIjp7XCJ4XCI6MCxcInlcIjowLFwid2lkdGhcIjo4NTAsXCJoZWlnaHRcIjoxMTAwfSxcInNlYXJjaFwiOnRydWUsXCJzaG93U3RhcnRTY3JlZW5cIjp0cnVlLFwiZ3JpZENvbG9yXCI6XCIjZDBkMGQwXCIsXCJkYXJrR3JpZENvbG9yXCI6XCIjNkU2RTZFXCIsXCJhdXRvc2F2ZVwiOnRydWUsXCJyZXNpemVJbWFnZXNcIjpudWxsLFwib3BlbkNvdW50ZXJcIjowLFwidmVyc2lvblwiOjE4LFwidW5pdFwiOjEsXCJpc1J1bGVyT25cIjpmYWxzZSxcInVpXCI6XCJcIn0iLCIuc2NyYXRjaHBhZCI6IjxteGxpYnJhcnk+W3tcInhtbFwiOlwialpEQkRvTWdESWFmcG5lRVpBOHd0M25hYVU5QVJpTWtJQWJyMUxjZlNqZXpnOGtPSk8zMzl5ZC9DNm9PYzVOMGIrL1JvQWQxQlZXbkdLbFVZYTdSZTVEQ0dWQVhrRkxrQi9KMm9GYWJLbnFkc0tOL0RMSVlYdHFQV0VnQkF5MmVRWXBqWjNDZHIwQ2RKK3NJSDcxK3J1cVVjMmRtS1hpVytUdE1oUE5ocEExeG5nWmpRRXBMSHBtY0ljc1RIRnRZZEsxbDI0bVpIa3JmZnEzN2dybmdIVC90ZnN0Tit6bjFHdz09XCIsXCJ3XCI6MTIwLFwiaFwiOjYwfSx7XCJ4bWxcIjpcImpaSEJic01nRElhZmhqdUZ2Y0NTYmozdHRMNEFDbDVBTXpnaWJwTysvWnpBVm0xU3BSMlE3TTgydG44cjI2ZjFWTndVM3NnREt2dWliRitJdUZwcDdRRlJHUjI5c2tkbGpKYW56T3VENkdHUDZza1Z5UHlmQWxNTHJnNHZVTWtaVnE1dzVoczJ5QnUwWGVBa0V4NFBZczVjNkJONlFpcENNbVhKN0Q0aTRoL2tNSTVaM0VFR0F1SGRGUXJId2VGekM2VG8vZGFtVzBKa2VKL2NzUFZjUkJGaGhTN1p3emFyRnEvTktoL0ErbkRmSGJWbFQwQUp1TndrWlltZVE4MTRxcExvQUhFTXJhckpwTjFjL2ZHbjhpNmVHRTIvYi9kK3B6MzI2NHhmXCIsXCJ3XCI6NDAsXCJoXCI6MjB9XTwvbXhsaWJyYXJ5PiJ9",
    "terminal.integrated.shell.osx": "/bin/zsh",
    "explorer.sortOrder": "type",
    "remote.SSH.remotePlatform": {
        "geo.bu.edu": "linux"
    },
    "python.dataScience.interactiveWindowMode": "perFile",
    "r.alwaysUseActiveTerminal": true,
    "terminal.integrated.inheritEnv": false
}
```

and here's the `keybindings.json`

```
// Empty
[
    {
        "key": "cmd+;",
        "command": "workbench.action.terminal.focus",
        "when": "editorFocus && !terminalFocus"
    },
    {
        "key": "cmd+;",
        "command": "workbench.action.focusActiveEditorGroup",
        "when": "!editorFocus && terminalFocus"
    },
    {
        "key": "shift+cmd+k",
        "command": "editor.action.deleteLines",
        "when": "textInputFocus && !editorReadonly"
    },
    {
        "key": "shift+cmd+k",
        "command": "-editor.action.deleteLines",
        "when": "textInputFocus && !editorReadonly"
    },
    {
        "key": "shift+cmd+enter",
        "command": "-r.runSourcewithEcho",
        "when": "editorTextFocus && editorLangId == 'r'"
    },
    {
        "key": "shift+cmd+s",
        "command": "-r.runSource",
        "when": "editorTextFocus && editorLangId == 'r'"
    },
    {
        "key": "cmd+6",
        "command": "r.runCommandWithSelectionOrWord",
        "when": "editorTextFocus && editorLangId == 'r'",
        "args": "str($$)"
    },
    {
        "key": "cmd+1",
        "command": "r.runCommandWithSelectionOrWord",
        "when": "editorTextFocus && editorLangId == 'r'",
        "args": "nrow($$)"
    },
    {
        "key": "cmd+2",
        "command": "r.runCommandWithSelectionOrWord",
        "when": "editorTextFocus && editorLangId == 'r'",
        "args": "length($$)"
    },
    {
        "key": "cmd+3",
        "command": "r.runCommandWithSelectionOrWord",
        "when": "editorTextFocus && editorLangId == 'r'",
        "args": "head($$)"
    },
    {
        "key": "cmd+4",
        "command": "r.runCommandWithSelectionOrWord",
        "when": "editorTextFocus && editorLangId == 'r'",
        "args": "tail($$)"
    },
    {
        "key": "cmd+5",
        "command": "r.runCommandWithSelectionOrWord",
        "when": "editorTextFocus && editorLangId == 'r'",
        "args": "names($$)"
    },
    {
        "key": "ctrl+backspace",
        "command": "deleteAllRight",
        "when": "textInputFocus && !editorReadonly"
    },
    {
        "key": "ctrl+k",
        "command": "-deleteAllRight",
        "when": "textInputFocus && !editorReadonly"
    },
    {
        "key": "alt+d",
        "command":"r.runCommandWithSelectionOrWord",
        "when": "editorTextFocus && editorLangId == 'r'",
        "args": "debugonce($$)"
    }
]
```
Thanks!

**Comment by andycraig** [comment]
I copied your `settings.json` and `keybindings.json` and tried to reproduce this but I couldn't. However, I'm trying on Linux. I want to check a few more things with you. If none of them solve it, hopefully someone with a Mac can investigate further.

1. By 'Remove the "insert new line" keybinding', do you mean 'Insert Line Below' (`editor.action.insertLineAfter`)? I can't find this in your `keybindings.json` file.

2. Here is how I tried to reproduce the issue. If there is anything different from what you are doing, could you let me know? 
    1. In a terminal, start radian
    1. Create a file `temp.R` with content `1 + 1`
    1. With cursor in this file, use <kbd>Ctrl+Enter</kbd> (since I'm using Linux)
    1. Observe the following is sent to the radian console:
        ```
        r$> 1 + 1
        [1] 2
        ```
    1. Close the terminal by clicking the trash icon
    1. Start a new terminal, and start radian
    1. With cursor in this file, use <kbd>Ctrl+Enter</kbd> (since I'm using Linux)
    1. I observe the following is sent to the radian console, but in the issue description the cursor moves to the next line and nothing is sent to the console
        ```
        r$> 1 + 1
        [1] 2
        ```

3. Can you try the following?
    1. Run command `Developer: Toggle Keyboard Shortcuts Troubleshooting`
    1. Do the steps I've described above in 2.
    1. Switch from 'TERMINAL' to 'OUTPUT' and copy-paste the last few lines here. For me the last few lines look like this:
        ```
        [2020-12-20 12:53:58.043] [renderer1] [info] [KeybindingService]: / Received  keydown event - modifiers: [ctrl], code: Enter, keyCode: 13, key: Enter
        [2020-12-20 12:53:58.043] [renderer1] [info] [KeybindingService]: | Converted keydown event - modifiers: [ctrl], code: Enter, keyCode: 3 ('Enter')
        [2020-12-20 12:53:58.043] [renderer1] [info] [KeybindingService]: | Resolving ctrl+[Enter]
        [2020-12-20 12:53:58.043] [renderer1] [info] [KeybindingService]: \ From 13 keybinding entries, matched r.runSelection, when: editorTextFocus && editorLangId == 'r', source: built-in extension Ikuyadeu.r.
        ```

Hopefully one of the above will provide some hints as to where the problem is.

**Comment by renkun-ken** [comment]
@MrJGao From your settings, I suspect that you have enabled both [Ikuyadeu.r](https://marketplace.visualstudio.com/items?itemName=Ikuyadeu.r) and [Mikhail-Arkhipov.r](https://marketplace.visualstudio.com/items?itemName=Mikhail-Arkhipov.r). The two extensions have some conflicts. Could you check if it is the case? If this is the case, then disabling Mikhail-Arkhipov.r should resolve the problem.

**Comment by MrJGao** [comment]
Hi, thanks for investigating this issue and sorry about my late reply. It is strange but `cmd+enter` seems working now.

1. I removed the `Insert Line Below` in the keyboard shortcut setting window (not the json file), it didn't work. But, when I changed the `when` condition, it worked. I changed when the condition of `Insert Line Below` to 

```editorTextFocus && !editorReadonly && editorLangId != 'r' ```

2. I said it's weird because now even if I change the when condition back to default, it also works. 

I'm not sure if VSCODE has updated or something happened, I couldn't reproduce the issue myself on both Macbook pro. Sorry, maybe we should close this issue, and if I get the issue again, I'll try your suggestions and figure it out.

Thank you.
